lib/filters.js

```diff
@@ -14,57 +14,74 @@ exports.tasks_by_id = function(doc, req) {
 };
 
 exports.doc_by_place = function(doc, req) {
-  var ok = function(place) {
-    if (!req.query.id) {
-      // admin
-      return true;
+  var ALL = "all",
+      UNASSIGNED = "unassigned",
+      ADMIN = "admin";
+
+  function usersThatCanSeeThisDoc(doc) {
+    var ok = function(place) {
```

Comment:
rename `ok` to `usersThatCanSeeThisContact`, and the arg is `contactOrFacility` (ugly yet accurate!)